app/models/game.rb

```diff
@@ -39,20 +39,16 @@ def determine_checkmate
   # rubocop:disable Metrics/MethodLength
   def checkmate(id)
     current_pieces = pieces.where(user_id: id, captured: false)
-    check_status = true
+    status = true
     current_pieces.each do |piece|
       8.times do |row|
         8.times do |col|
-          next unless piece.valid_move?(row, col)
-          Piece.transaction do
-            piece.try_to_move(row, col)
-            check_status = false if determine_check == false
-            fail ActiveRecord::Rollback
-          end
+          next unless piece.reload.valid_move?(row, col)
```

Comment:
Look at line 94 -- the `checkmate_check` method.  You'll see what you're looking for.